### PR TITLE
Refx/tweaks

### PR DESCRIPTION
**Pull request by FigBug**
- Window now saves it's position
- Window can be deleted without leaking listeners
- Clean up warnings

### CODE REVIEW

File: melatonin_inspector.h [hunk 1]
```diff
@@ -66,21 +66,77 @@ namespace melatonin
...
+        {
+            if (settings_ != nullptr)
+            {
+                settings.set (settings_, false);
+            }
+            else
+            {
+                juce::PropertiesFile::Options opts;
+
+                opts.applicationName = "refx";
```
Comment (sudara) [inline melatonin_inspector.h:87]: should we change this to "melatonin_inspector"?

File: melatonin/helpers/misc.h [hunk 2]
```diff
@@ -1,36 +1,37 @@
...
-            return juce::ImageCache::getFromMemory (data, size);
-        }
-        else
-        {
-            // you told me to load an image that doesn't exist
-            jassertfalse;
-        }
-    }
-
-}
```
Comment (sudara) [inline melatonin/helpers/misc.h:36]: did this file have some line ending issues ? just curious about the diff...

File: melatonin/helpers/misc.h [hunk 3]
```diff
@@ -1,36 +1,37 @@
...
+        auto data = InspectorBinaryData::getNamedResource (filename.getCharPointer(), size);
+        if (size > 0)
+        {
+            return juce::ImageCache::getFromMemory (data, size);
+        }
+        else
+        {
+            // you told me to load an image that doesn't exist
+            jassertfalse;
+            return {};
```
Comment (sudara) [inline melatonin/helpers/misc.h:33]: thanks for catching this, I must not have warnings set on macOS for lack of return...

**Review by sudara — COMMENTED** [review]
Awesome, thanks for catching this stuff. Seems I need better warnings in my modules. 

Just let me know if we can change the applicationName and I'll merge.

File: melatonin_inspector.h (hunk 1 above)
Comment (FigBug) [reply melatonin_inspector.h:87]: Yes, oops. Copy / pasted from some other code

File: melatonin/helpers/misc.h (hunk 2 above)
Comment (FigBug) [reply melatonin/helpers/misc.h:36]: Not sure what happened there, maybe xcode changed the line endings? What line ending do you prefer?

Comment (sudara) [reply melatonin/helpers/misc.h:36]: I have them set to Unix/macOS `\n` but had some trouble in the past, so it's possible it got cleaned up here... github doesn't make it easy to see....

**Comment by sudara** [comment]
Thanks! Much appreciated and happy to see it being used (I use it almost daily!)

Feel free to open issues for any issues/features...

**Comment by sudara** [comment]
Testing this locally,  there's a few regressions in the UI behavior.

1. The inspector's first open was unideal (enabled, but not showing properties, which shouldn't be possible):

AudioPluginHost - 2023-08-02 18@2x

When disabling the inspector, it the behavior was to contract back to its disabled small floating window state:

AudioPluginHost - 2023-08-02 16@2x

After this PR, disabling does something like this:
AudioPluginHost - 2023-08-02 15@2x

**Comment by sudara** [comment]
I've got a PR that restores the original behavior, but before I merge could you let me know what the purpose of the `onClose` callback is? 

My guess is that you don't care about the small "disabled" state and you are toggling open/close the DocumentWindow from within your app. Is that right? If so, I wonder if i better solution is just to remove the `toggle(false)` from `closeButtonPressed`...

**Comment by FigBug** [comment]
I don't keep the melatonin::Inspector around all the time. I create it when need it and then delete it when the user closes the window. I use the onClose callback to know when to delete the component.

**Comment by sudara** [comment]
Can you share your onClose callback with me? Just want to make sure I have the bases covered here...

**Comment by sudara** [comment]
Er, actually, it would be useful to see how you both open and close the inspector.... For example, are you calling `addAndMakeVisible` on open?

**Comment by FigBug** [comment]
```
	debugMenu.addItem ( "show component inspector", true, editor.componentInspector != nullptr, [ this ]
	{
		if ( editor.componentInspector == nullptr )
		{
			editor.componentInspector = std::make_unique<melatonin::Inspector> ( editor );
			editor.componentInspector->setVisible ( true );
			editor.componentInspector->toggle ( true );
			editor.componentInspector->onClose = [ this ] { editor.componentInspector = nullptr; };
			editor.componentInspector->setLookAndFeel ( &getLookAndFeel () );
		}
		else
		{
			editor.componentViewer = nullptr;
		}
	} );
```

**Pull request merged**